### PR TITLE
swift-inspect: clean up event creation on Windows

### PR DESCRIPTION
**Pull request by compnerd**
Create a Swiftier interface for `CreateEvent` from the Windows SDK and use that to simplify the implementation of the remote process interface.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @hjyamauchi

File: tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift [hunk 1]
```diff
@@ -529,30 +529,11 @@ internal final class WindowsRemoteProcess: RemoteProcess {
...
-      false,  // Initial state is nonsignaled
-      writeEventName)
-    if hWriteEvent == HANDLE(bitPattern: 0) {
-      print("CreateEvent failed \(GetLastError())")
-      CloseHandle(hReadEvent)
-      return nil
-    }
+    let hReadEvent = CreateEvent("\(READ_EVENT_NAME_PREFIX)-\(dwProcessId)")
+    let hWriteEvent = CreateEvent("\(WRITE_EVENT_NAME_PREFIX)-\(dwProcessId)")
+    guard let hReadEvent, let hWriteEvent else { return nil }
```
Comment (mikeash) [inline tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift:534]: The new code misses the `CloseHandle(hReadEvent)` when `hWriteEvent` fails.

**Review by mikeash — APPROVED** [review]
Much nicer.

File: tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift (hunk 1 above)
Comment (compnerd) [reply tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift:534]: Right you are sir, well spotted.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by hjyamauchi** [comment]
Nice

**Pull request merged**